wdfn-server/waterdata/__init__.py

```diff
@@ -43,18 +47,100 @@ def _create_log_handler(log_directory=None, log_name=__name__):
+            app.config[lookup_name] = json.loads(f.read())
+            if not is_first_load:
+                app.logger.debug(f'Successfully loaded old lookup file for {lookup_name}.json, '
+                                 'but will send email to team as an update.')
+                message = f'The WDFN application could not get the lookup file for {lookup_name}.json ' \
+                          'but was able to load a backup file. This may be caused by an intermittent connection ' \
+                          'problem and should not affect the function of the application. However, if you receive ' \
+                          'more of these emails, an investigation is in order.'
+                send_email(message)
+    except FileNotFoundError:
+        app.logger.error('Sorry, no previous file found; prepare for the worst.')
+        if not is_first_load:
+            app.logger.error('Sorry, no previous file found. Sending email message for help.')
+            message = f'IMPORTANT - The WDFN application could not get the lookup file for {lookup_name}.json,' \
+                      ' and could not load an old copy. This will cause the application to fail. ' \
+                      'Send help soon!'
+            send_email(message)
+
+
+# Pull lookup files from S3 bucket and load into application context
+def get_lookups():
+    """
+    Makes requests to an AWS Simple Storage Solutions Bucket (S3) to get various 'lookup' files
+    :return: None
+    """
+    if not os.path.exists(os.path.join(app.config.get('DATA_DIR'), 'lookups')):
+        os.makedirs(os.path.join(app.config.get('DATA_DIR'), 'lookups'))
+
+    for lookup in app.config['LOOKUP_ENDPOINTS']:
+        app.logger.debug('Getting lookup from {} '.format(app.config['LOOKUP_ENDPOINTS'].get(lookup)))
+
+        try:
+            request = requests.get(app.config['LOOKUP_ENDPOINTS'].get(lookup))
+            app.config[lookup] = request.json()
+            output_file_path = os.path.join(app.config.get('DATA_DIR'), 'lookups/{}.json'.format(lookup.lower()))
+            with open(output_file_path, 'w') as output_file:
+                json.dump(request.json(), output_file)
+
+        except ValueError as e:
+            app.logger.error('No Lookup JSON returned, failed with error {}'.format(e))
+            load_lookup_from_backup_file(lookup, is_first_load = False)
+        except requests.exceptions.RequestException as e:
+            app.logger.error('Request to get lookup file failed for {} with error: {}  '.format(lookup, e))
+            load_lookup_from_backup_file(lookup, is_first_load = False)
+
+
+# When the application is first started, there will be no backup files saved for the lookups and the lookups
+# will not be in memory. So, let's get them.
+# If the backup lookup files already exist, then we can skip this step.
+# NOTE: The lookup files are saved locally but not committed to Git. This stops the application from pulling the
```

Comment:
I am not opposed to doing this, but I think it is rare that the application is restarted without a redeployment. Occassionally this happens if we update the config.  But I think it is fine to leave it in.